### PR TITLE
[1862] fix presidency check when selling shares

### LINKED ISSUE #8901: 1862 President sale of share
In [game 113371](https://18xx.games/game/113371?action=435), I was President of the L&H in SR4.  It was valued at 220.  I sold a share, and the value did not drop.  My understanding is that when the President sells shares, the price drops regardless of the zone.

https://gist.github.com/benjaminxscott/341a7dac1792d3abab743a5677964399

### PR DESCRIPTION
**Pull request by benjaminxscott**
fixes #8901

### CODE REVIEW

**Comment by HogueEric** [comment]
Thank you.

**Comment by roseundy** [comment]
@benjaminxscott 

I'm just catching up on this. Can you explain why bundle.owner doesn't work here?

File: lib/engine/game/g_1862/game.rb [hunk 1]
```diff
@@ -1196,7 +1196,7 @@ def sell_shares_and_change_price(bundle, allow_president_change: true, swap: nil
 
           @share_pool.sell_shares(bundle, allow_president_change: allow_president_change, swap: swap)
           num_shares = bundle.num_shares
-          unless corporation.owner == bundle.owner
+          unless corporation.owner == bundle.president
```
Comment (roseundy) [inline lib/engine/game/g_1862/game.rb:1199]: What am I missing? Isn't bundle.president defined to be corporation.owner? Won't this always be true?

**Comment by roseundy** [comment]
Can we please revert this? This is not the correct fix for the bug. This will break games.

File: lib/engine/game/g_1862/game.rb (hunk 1 above)
Comment (benjaminxscott) [reply lib/engine/game/g_1862/game.rb:1199]: bundle.owner points to the sharepool object that the bundle is in, afaik

Comment (roseundy) [reply lib/engine/game/g_1862/game.rb:1199]: My point is that bundle.president always points to corporation.owner, regardless of who owns the bundle. This means that the price will always drop even if it's not the president selling the shares.

The bug is that I don't make this check until after calling sell_shares, and thus the price never drops for one or two shares.

Comment (benjaminxscott) [reply lib/engine/game/g_1862/game.rb:1199]: Ah I see what you're saying

Further context to my above comment which I think was part of my confusion, `bundle.owner` in the bugged implementation returned a `SharePool` object after `sell_shares`, so the comparison was silently failing

https://github.com/tobymao/18xx/issues/8916

**Pull request merged**